spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/SimpleUrlHandlerMappingGlobalCorsAutoConfiguration.java

```diff
@@ -0,0 +1,34 @@
+package org.springframework.cloud.gateway.config;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping;
+
+/**
+ * This is useful for PreFlight CORS requests. We can add a "global" configuration here so
+ * we don't have to modify existing predicates to allow the "options" HTTP method.
+ * 
+ */
+
+@Configuration
+@ConditionalOnClass(SimpleUrlHandlerMapping.class)
+@ConditionalOnProperty(name = "spring.cloud.gateway.globalcors.addToSimpleUrlHanderMapping", matchIfMissing = true)
```

Comment:
Please use the dash case for property names. I think this should be opt in, not opt out. It should also be documented.

Reply:
@spencergibb updated.